starlette/middleware/base.py

```diff
@@ -144,7 +144,7 @@ async def send_no_error(message: Message) -> None:
             async def coro() -> None:
                 nonlocal app_exc
 
-                async with send_stream:
+                async with send_stream, recv_stream:
```

Comment:
fyi - you can use a syncronous with with memory object streams

Reply:
probably you want the with on line 112:

```python
send_stream, recv_stream = anyio.create_memory_object_stream()
with send_stream, recv_stream:
    ...
    task_group.start_soon(coro, send_stream.clone())
```

Reply:
Is the current approach on this PR wrong, or can I merge it?

Reply:
I'm not understanding why you have to catch ClosedResourceError here

Reply:
Yep, proposal from @graingert makes more sense than initial one from me: it's better to ensure cleanup with a context manager right away + `coro` doesn't really work with `recv_stream`, so it's weird to handle it where I originally put it

One thing that stops me from finalizing is a deadlock in some tests (e.g., `test_fully_evaluated_response`) that I don't quite see sources of 🤨